src/ansys/dpf/composites/_composite_model_impl.py

```diff
@@ -0,0 +1,618 @@
+        if time_in:
+            scope_config.add("requested_times", time_in)
+
+        scope_config_reader_op.inputs.scope_configuration(scope_config)
+
+        # configure operator to chunk the scope
+        chunking_config: Dict[str, Union[int, Sequence[str], Sequence[float]]] = {
+            "max_chunk_size": max_chunk_size
+        }
+        if ns_in:
+            chunking_config["named_selections"] = ns_in
+        chunking_data_tree = dpf.DataTree(chunking_config)
+
+        chunking_generator = dpf.Operator("composite::scope_generator")
+        chunking_generator.inputs.stream_provider(self._get_rst_streams_provider())
+        chunking_generator.inputs.data_tree(chunking_data_tree)
+        chunking_generator.inputs.data_sources(self.data_sources.composite)
+        if element_scope_in:
+            element_scope = dpf.Scoping(location="elemental")
+            element_scope.ids = element_scope_in
+            chunking_generator.inputs.element_scoping(element_scope)
+
+        min_merger = dpf.Operator("merge::fields_container")
+        max_merger = dpf.Operator("merge::fields_container")
+
+        merge_index = 0
+
+        while True:
+            chunking_generator.inputs.generator_counter(merge_index)
+            finished = chunking_generator.outputs.is_finished()
+            if finished:
+                break
+
+            evaluate_failure_criterion_per_scope_op = dpf.Operator(
+                "composite::evaluate_failure_criterion_per_scope"
+            )
+
+            evaluate_failure_criterion_per_scope_op.inputs.criterion_configuration(
+                combined_criterion.to_json()
+            )
+
+            evaluate_failure_criterion_per_scope_op.inputs.scope_configuration(
+                scope_config_reader_op.outputs
+            )
+
+            evaluate_failure_criterion_per_scope_op.inputs.element_scoping(
+                chunking_generator.outputs
+            )
+            evaluate_failure_criterion_per_scope_op.inputs.materials_container(
+                self.material_operators.material_provider.outputs
+            )
+            evaluate_failure_criterion_per_scope_op.inputs.stream_provider(
+                self._get_rst_streams_provider()
+            )
+            evaluate_failure_criterion_per_scope_op.inputs.mesh(self.get_mesh())
+            evaluate_failure_criterion_per_scope_op.inputs.has_layup_provider(has_layup_provider)
+            evaluate_failure_criterion_per_scope_op.inputs.section_data_container(
+                self._layup_provider.outputs.section_data_container
+            )
+            evaluate_failure_criterion_per_scope_op.inputs.material_fields(
+                self._layup_provider.outputs.material_fields
+            )
+            evaluate_failure_criterion_per_scope_op.inputs.mesh_properties_container(
+                self._layup_provider.outputs.mesh_properties_container
+            )
+            # always set to true to ensure that solid stacks are grouped
```

Comment:
Is this comment correct? request_sandwich_results=false will just skip the sandwich criteria.